client/apps/shipping-label/index.js

```diff
@@ -30,14 +30,64 @@ import initializeLabelsState from 'woocommerce/woocommerce-services/lib/initiali
 import './style.scss';
 
 export default ( { order, accountSettings, packagesSettings, shippingLabelData, continents, context, items } ) => {
-
 	const orderId = order ? order.id : null;
-	const { storeOptions, formMeta, userMeta, formData } = accountSettings;
-	const packages = packagesSettings.formData;
-	const dimensionUnit = packagesSettings.storeOptions.dimension_unit;
-	const weightUnit = packagesSettings.storeOptions.weight_unit;
-	const packageSchema = packagesSettings.formSchema.custom ? packagesSettings.formSchema.custom.items : undefined;
-	const predefinedSchema = packagesSettings.formSchema.predefined;
+	let preLoadedState = false;
+	if ( undefined !== accountSettings ) {
+		preLoadedState = true;
+	}
+
+	const addPreloadedState = function( initialState ) {
+		const { storeOptions, formMeta, userMeta, formData } = accountSettings;
+		const packages = packagesSettings.formData;
+		const dimensionUnit = packagesSettings.storeOptions.dimension_unit;
+		const weightUnit = packagesSettings.storeOptions.weight_unit;
+		const packageSchema = packagesSettings.formSchema.custom ? packagesSettings.formSchema.custom.items : undefined;
+		const predefinedSchema = packagesSettings.formSchema.predefined;
+		initialState.extensions.woocommerce.woocommerceServices = {
```

Comment:
It looks like `initialState` (which is an object passed by reference in the function's argument) is being altered directly (rather than either being initialized in the function's body or being cloned).

I haven't been able to see any direct side effects related to this decision, but I just wanted to point it out and double check if this could be causing any issues?

Reply:
Since the variable is initialized in `getInitialState` and immediately modified, I don't personally see a risk here.